データベース設計

kumazakiz opened the pull request:
#WHAT
ER図作成とREADEMEに記述をする。

#WHY
モデルの設計とアソシエーションの設定を明確にするため。

ER図の内容
https://gyazo.com/1c8f472651a7f356ec0299914567ae07

kumazakiz committed:
gem rails_12factorの追加

kumazakiz committed:
ER図作成

kumazakiz committed:
DB設計(1)

kumazakiz committed:
DB設計(2)

・ER図設計の修正情報も入っています。

Katsuno0814 commented on README.md:
```diff
@@ -1,24 +1,68 @@
+| family_name_kana | string | null: false |
+| first_name_kana  | string | null: false |
+| birthday         | string | null: false |
 
```
今回の実装では、誕生日カラムは「date型」で設計した方が実装が簡単になると考えます。（date型は年、月、日を1つで指定して取得できるためです！）

Katsuno0814 commented on README.md:
```diff
@@ -1,24 +1,68 @@
+| ------------- | ---------- | ------------------------------ |
+| title         | string     | null: false                    |
+| explanation   | text       | null: false                    |
+| category      | string     | null: false                    |
+| item_status   | string     | null: false                    |
+| shipping_cost | string     | null: false                    |
+| shipping_area | string     | null: false                    |
+| days_to_ship  | string     | null: false                    |
```
**商品の状態, 配送料の負担, 発送元の地域,  発送までの日数, カテゴリー**のカラムはactive_hashで実装の予定のため、integer型にしましょう！
詳しくは下記のカリキュラムを参考にしてください！（references型は使用しません。）
https://master.tech-camp.in/curriculums/4753

Katsuno0814 commented on README.md:
```diff
@@ -1,24 +1,68 @@
+### Association
 
-* How to run the test suite
+- belongs_to :buyer
```
今回の実装では「1つの商品は1つの購入履歴を持っている」という設計にしたいため、has_oneを使用してあげましょう！

Katsuno0814 commented on README.md:
```diff
@@ -1,24 +1,68 @@
+| ------------- | ---------- | ------------------------------ |
+| user          | references | null: false, foreign_key: true |
+| item          | references | null: false, foreign_key: true |
+| purchase_date | datetime   | null: false                    |
```
商品の購入管理はuserとitemのidのみで管理が可能です。
したがって、purchase_dateカラムは今回は不要です！

Katsuno0814 commented on README.md:
```diff
@@ -1,24 +1,68 @@
+
+### Association
+
+- has_many :items
+- belongs_to :user
```
belongs_to :itemに修正いたしましょう！

Katsuno0814 commented on README.md:
```diff
@@ -1,24 +1,68 @@
+
+| Column        | Type       | Options                        |
+| ------------- | ---------- | ------------------------------ |
+| postal_code   | integer    | null: false                    |
```
郵便番号はハイフンを必要とすることも考慮してstring型で実装しましょう！

Katsuno0814 commented on README.md:
```diff
@@ -1,24 +1,68 @@
+| Column        | Type       | Options                        |
+| ------------- | ---------- | ------------------------------ |
+| postal_code   | integer    | null: false                    |
+| prefectures   | string     | null: false                    |
```
こちら都道府県のカラムはactive_hashで実装の予定のため、integer型にしましょう！
詳しくは下記のカリキュラムを参考にしてください！
最終課題の補足カリキュラム「1つのフォームから複数のテーブルに情報を保存するアプリを作成しよう」
をご参照ください。

Katsuno0814 commented on README.md:
```diff
@@ -1,24 +1,68 @@
+| municipality  | string     | null: false                    |
+| address       | string     | null: false                    |
+| building_name | string     |                                |
+| phone_number  | integer    | null: false                    |
```
電話番号は数字の文字列として実装しておきたいためstring型で実装しましょう！

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認をお願いいたします！

kumazakiz committed:
ER図・READEMEの修正

Katsuno0814 commented on README.md:
```diff
@@ -20,47 +20,46 @@
+| title         | string  | null: false |
+| explanation   | text    | null: false |
+| category      | string  | null: false |
+| item_status   | integer | null: false |
+| shipping_cost | integer | null: false |
+| shipping_area | integer | null: false |
+| days_to_ship  | integer | null: false |
```
こちら後々のエラーの原因になってしまう可能性があるので指摘させていただきます。
active_hashで導入するこれらのカラム名は語尾に「_id」をつけましょう。
例：category_id

Katsuno0814 commented on README.md:
```diff
@@ -20,47 +20,46 @@
+| shipping_area | integer | null: false |
+| days_to_ship  | integer | null: false |
+| selling_price | integer | null: false |
+| user          | string  | null: false |
```
こちら
`| user          | references | null: false, foreign_key: true |`
といたしましょう！

Katsuno0814 commented on README.md:
```diff
@@ -20,47 +20,46 @@
-| postal_code   | integer    | null: false                    |
-| prefectures   | string     | null: false                    |
+| postal_code   | string     | null: false                    |
+| prefectures   | integer    | null: false                    |
```
`| prefectures   | integer    | null: false                    |`

こちらactive_hashを使用した場合、カラム名に「_id」をつけてあげましょう。「_id」がない場合、後にエラーになってしまう可能性が高くなります。

Katsuno0814 commented on README.md:
```diff
@@ -20,47 +20,46 @@
 | building_name | string     |                                |
-| phone_number  | integer    | null: false                    |
+| phone_number  | string     | null: false                    |
 
```
|buyer         | prefectures   | integer    | null: false                    |
を追加いたしましょう！

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

kumazakiz committed:
ER図・READEMEの修正(2)

READEME.mdの修正指示「|buyer | prefectures | integer | null: false |
を追加いたしましょう！」はshipping_addresses テーブルにカラム名：buyer、カラム型：integer、オプション：null:falseの追加と解釈して記述しました。

JUNSHIROBABA reviewed (commented):
レビュー依頼ありがとうございます。
２点程追加でコメントさせて頂きました。確認の程よろしくお願いいたします。

完成までもう少しです！

細かい点を指摘しておりますが、データベース設計はコントローラーやモデル、テーブル作成の指標となるとても重要なものです。
ここをしっかりと作成しておくことにより、後の作業を円滑に進めることができます。

何卒ご理解頂けますようよろしくお願いいたします。

kumazakiz committed:
ER図・READEMEの修正(3)

JUNSHIROBABA reviewed (commented):
レビュー依頼ありがとうございます。

こちらはLGTMとさせていただきます！長らくのデータベース設計の実装、たいへんお疲れ様でした！

引き続き、次の機能の実装を進めていきましょう！
